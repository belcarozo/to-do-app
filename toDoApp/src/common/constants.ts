import { routesType } from '../navigation/types'

export const constants = {
  clear: 'CLEAR ALL DONE',
  homeTitle: 'Todo',
  detailsTitle: 'Details',
  newTaskTitle: 'New task',
  done: 'Done',
  notDone: 'notDone',
  add: '+',
  save: 'Save',
  cancel: 'Cancel',
  taskTitle: 'Task title',
  taskDescription: 'Task description',
}

export const routes: routesType = {
  home: 'Home',
  newTask: 'NewTask',
  details: 'Details',
}
