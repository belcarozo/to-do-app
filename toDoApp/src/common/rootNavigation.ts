import { NavigationContainerRef, ParamListBase } from '@react-navigation/native'
import React from 'react'

export const navigationRef: React.RefObject<NavigationContainerRef> = React.createRef()

export const goBack = () => {
  navigationRef.current?.goBack()
}

export const navigate = (name: string, params: ParamListBase) => {
  navigationRef.current?.navigate(name, params)
}
