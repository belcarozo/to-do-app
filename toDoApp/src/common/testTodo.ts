export const todo = { title: 'test', description: 'test', completed: true }
export const todoid = {
  title: 'test',
  description: 'test',
  completed: true,
  id: 0,
}
