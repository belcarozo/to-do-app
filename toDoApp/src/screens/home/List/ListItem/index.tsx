import React from 'react'
import { View, Text, Pressable } from 'react-native'
import { styles } from './styles'
import { Check } from './Check/index'
import { ListItemInterface } from './types'

export const ListItem: React.FC<ListItemInterface> = ({
  title,
  description,
  completed,
  id,
  onPress,
  onPressTitle,
}) => {
  return (
    <View style={styles.container}>
      <Pressable
        style={styles.pressableTitle}
        onPress={onPressTitle(title, description, completed, id)}>
        <View style={styles.data}>
          <View style={styles.textSection}>
            <Text style={styles.titleFormat}>{title}</Text>
            <View style={styles.description}>
              <Text style={styles.descriptionFormat}>{description}</Text>
            </View>
          </View>
          <View style={styles.checkboxSection}>
            <Check
              title={title}
              description={description}
              onPress={onPress}
              id={id}
              completed={completed}
            />
          </View>
        </View>
        <View style={styles.separator} />
      </Pressable>
    </View>
  )
}
