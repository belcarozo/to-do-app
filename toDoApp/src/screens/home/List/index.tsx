import React from 'react'
import { FlatList } from 'react-native'
import { ListItem } from './ListItem'

const DATA = [
  {
    title: 'Buy groceries',
    subtitle: 'Fruit and vegetables',
  },
  {
    title: 'Clean bathroom',
    subtitle: 'Wash towels',
  },
  {
    title: 'Send e-mail to e-bay',
    subtitle: 'Package was never delivered',
  },
  {
    title: 'Buy new mouse',
    subtitle: 'Old one is broken',
  },
  {
    title: 'Take out the trash',
    subtitle: 'Before the smell gets worse',
  },
]

const renderItem = ({ item }) => (
  <ListItem title={item.title} subtitle={item.subtitle} />
)

export const List = () => <FlatList data={DATA} renderItem={renderItem} />
