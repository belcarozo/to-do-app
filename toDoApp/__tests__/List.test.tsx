import React from 'react'
import { shallow } from 'enzyme'
import { todoid } from '../src/common/testTodo'
import { List } from '../src/screens/Home/List'

describe('Rendering list', () => {
  it('should match to snapshot', () => {
    const component = shallow(<List list={[todoid]} renderItem={jest.fn()} />)
    expect(component).toMatchSnapshot()
  })
})
