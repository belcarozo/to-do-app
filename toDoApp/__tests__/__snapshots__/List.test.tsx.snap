// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering list should match to snapshot 1`] = `
<FlatList
  data={
    Array [
      Object {
        "completed": true,
        "description": "test",
        "id": 0,
        "title": "test",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[MockFunction]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
